### PR TITLE
oneDAL kmeans_init hot fix

### PR DESCRIPTION
**Pull request by mchernov-intel**
- kmeans_init::task
- build failure fix 
- other minor changes 

hot_fix

### CODE REVIEW

**Comment by mchernov-intel** [comment]
@rlnx please, review

**Comment by andreynv** [comment]
on my side I do not have other comments

**Review by andreynv — APPROVED** [review]

File: source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp [hunk 1]
```diff
@@ -78,7 +96,7 @@ class compute_result {
 /// @post :expr:`result.centroids.has_data == true`
 /// @post :expr:`result.centroids.row_count == desc.cluster_count`
 /// @post :expr:`result.centroids.column_count == input.data.column_count`
-template <typename Float, typename Method>
+template <typename Float, typename Method, typename Task>
 compute_result compute(const descriptor<Float, Method>& desc,
```
Comment (rlnx) [inline source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp:100]:
```suggestion
compute_result<Task> compute(const descriptor<Float, Method, Task>& desc,
```

File: source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp [hunk 2]
```diff
@@ -78,7 +96,7 @@ class compute_result {
...
                    const compute_input& input);
```
Comment (rlnx) [inline source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp:101]:
```suggestion
                   const compute_input<Task>& input);
```

File: source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp [hunk 3]
```diff
@@ -11,18 +11,29 @@ namespace method {
    using by_default = dense;
 } // namespace method
 
+namespace task {
+   /// Tag-type that parametrizes entities used for solving
+   /// :capterm:`initialization problem <init>`.
```
Comment (rlnx) [inline source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp:16]:
```suggestion
   /// Tag-type that parametrizes entities used for obtaining the initial K-Means centroids.
```

File: source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp (hunk 1 above)
Comment (mchernov-intel) [reply source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp]: fixed

File: source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp (hunk 2 above)
Comment (mchernov-intel) [reply source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp]: fixed

File: source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp (hunk 3 above)
Comment (mchernov-intel) [reply source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp]: fixed

**Review by rlnx — APPROVED** [review]

**Pull request merged**